triggers.lesson14/styles-for-triggers.css

```diff
@@ -0,0 +1,165 @@
+.container div {
+    display: flex;
+    margin: 150px;
+    padding: 10px 150px 10px 10px;
+    border: rgb(7, 69, 151) solid 2px;
+}
+
+.content div {
+    font-size: 20px;
+    padding: 0px;
+    border: 2px solid rgb(201, 63, 139);
+    margin: 10px;
+}
+
+.container {
+    position: absolute;
+    animation: slide-from-left 1s;
+    animation-duration: 3s;
+    animation-timing-function: ease-out;
+}
+
+@keyframes slide-from-left {
+    from {
+        left: -1000px;
+    }
+
+    to {
+        left: 0;
+    }
+}
+
+.tooltip {
+    position: relative;
+    display: inline-block;
+    cursor: default;
+}
+
+.tooltip .tooltip-top {
+    width: 150px;
+    visibility: hidden;
+    position: absolute;
+    border-radius: 5px;
+    background-color: silver;
+    padding: 5px;
+    text-align: center;
+    bottom: 160%;
+    left: -40%;
+    opacity: 0;
+    transition: opacity 0.7s, visibility 0.7s;
+}
+
+.tooltip:hover .tooltip-top{
+    visibility: visible;
+    opacity: 1;
+}
+
+.tooltip-top::before {
+    content: "";
+    width: 0px;
+    height: 0px;
+    position: absolute;
+    left: 45%;
+    top: 100%;
+    text-align: center;
+    border: 10px solid transparent;
+    border-top-color: silver;  
+}
+
+.tooltip .tooltip-right {
+    width: 150px;
```

Comment:
As you see you have a lot of common styles for all tooltips. 
So I suggest you to set common styles to ".tooltip span" and only specific class to "tooltip-right", "tooltip-left" ...

Reply:
I use opacity for the smooth effect, but I need hidden so that the tooltip does not appear when the cursor is not on the trigger.
Now I'm trying to get a smooth effect from hidden only and I can't...